### PR TITLE
[2430] Course study mode question - sending to DTTP

### PR DESCRIPTION
**Pull request by mooktakim**
### Context

https://trello.com/c/abUO5hnT/2430-m-course-study-mode-question-sending-to-dttp

### Changes proposed in this pull request

* `dfe_studymodeid@odata.bind` param added to `Dttp::Params::PlacementAssignment`

### CODE REVIEW

**Comment by mooktakim** [comment]
[[M] Course study mode question - sending to DTTP](https://trello.com/c/abUO5hnT/2430-m-course-study-mode-question-sending-to-dttp)

File: app/lib/dttp/params/placement_assignment.rb [hunk 1]
```diff
@@ -118,6 +119,14 @@ def training_initiative_param
         }
       end
 
+      def study_mode_params
+        return {} if trainee.study_mode.blank?
+
+        {
+          "dfe_studymodeid@odata.bind" => "/dfe_studymodeses(#{course_study_mode_id(trainee.study_mode)})",
```
Comment (darokel) [inline app/lib/dttp/params/placement_assignment.rb:126]: If this is an `odata.bind` i think the format of the params needs to be `dfe_StudyModeId`

File: spec/lib/dttp/params/placement_assignment_spec.rb [hunk 2]
```diff
@@ -378,6 +378,50 @@ module Params
...
+                provider: provider,
+                study_mode: COURSE_STUDY_MODES[:full_time],
+              )
+            end
+
+            subject { described_class.new(trainee).params }
+
+            it "sets study_mode" do
+              expect(subject).to include({
+                "dfe_studymodeid@odata.bind" => "/dfe_studymodeses(#{CodeSets::CourseStudyModes::MAPPING[COURSE_STUDY_MODES[:full_time]][:entity_id]})",
```
Comment (darokel) [inline spec/lib/dttp/params/placement_assignment_spec.rb:400]: https://github.com/DFE-Digital/register-trainee-teachers/pull/1275/files#r690234761

File: spec/lib/dttp/params/placement_assignment_spec.rb [hunk 3]
```diff
@@ -378,6 +378,50 @@ module Params
...
+                dttp_id: dttp_contact_id,
+                provider: provider,
+                study_mode: nil,
+              )
+            end
+
+            subject { described_class.new(trainee).params }
+
+            it "does not set study_mode" do
+              expect(subject.keys).not_to include("dfe_studymodeid@odata.bind")
```
Comment (darokel) [inline spec/lib/dttp/params/placement_assignment_spec.rb:421]: https://github.com/DFE-Digital/register-trainee-teachers/pull/1275/files#r690234761

**Review by darokel — APPROVED** [review]

**Pull request merged**